src/main/kotlin/app/revanced/patches/twitter/misc/links/ChangeLinkSharingDomainPatch.kt

```diff
@@ -0,0 +1,140 @@
+package app.revanced.patches.twitter.misc.links
+
+import app.revanced.extensions.exception
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.addInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.getInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.removeInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchException
+import app.revanced.patcher.patch.annotation.CompatiblePackage
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patcher.patch.options.PatchOption.PatchExtensions.stringPatchOption
+import app.revanced.patches.twitter.misc.links.fingerprints.LinkBuilderFingerprint
+import app.revanced.patches.twitter.misc.links.fingerprints.LinkResourceGetterFingerprint
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.builder.BuilderInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.TwoRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+
+
+@Patch(
+    name = "Change link sharing domain",
+    description = "Replaces the domain name of Twitter links when sharing them.",
+    compatiblePackages = [CompatiblePackage("com.twitter.android")]
+)
+@Suppress("unused")
+object ChangeLinkSharingDomainPatch : BytecodePatch(
+    setOf(LinkBuilderFingerprint, LinkResourceGetterFingerprint)
+) {
+    private var domainName by stringPatchOption(
+        key = "domainName",
+        default = "fxtwitter.com",
+        title = "Domain name",
+        description = "The domain to use when sharing links.",
+        required = true
+    )
+
+    override fun execute(context: BytecodeContext) {
+        val linkBuilderResult = LinkBuilderFingerprint.result ?: throw LinkBuilderFingerprint.exception
+
+        // region Copy link button.
+        linkBuilderResult.mutableMethod.apply {
+            val stringIndex = linkBuilderResult.scanResult.stringsScanResult!!.matches
+                .first().index
+
+            val targetRegister = getInstruction<OneRegisterInstruction>(stringIndex).registerA
+            replaceInstruction(
+                stringIndex,
+                "const-string v$targetRegister, \"https://$domainName/%1\$s/status/%2\$d\""
+            )
+        }
```

Comment:
Please review if the changes I pushed still work.

Reply:
This seems to work fine